src/amberscript/parser_pipeline_set_test.cc

```diff
@@ -0,0 +1,216 @@
+// Copyright 2019 The Amber Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or parseried.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "gtest/gtest.h"
+#include "src/amberscript/parser.h"
+
+namespace amber {
+namespace amberscript {
+
+using AmberScriptParserTest = testing::Test;
+
+TEST_F(AmberScriptParserTest, OpenCLSetMissingKernel) {
```

Comment:
Should we test that SET is called on an OPENCL-C shader type?

Reply:
The way the buffers are generated, the information in the SET is ignored if the shader is not an OPENCL-C shader. Would you prefer an error to that behaviour?

Reply:
My concern is that you either set the shader type wrong, or are trying to use set for something it isn't designed for at this point. So, even though it will do nothing, I think it would be useful to generate an error to say something along the lines of "SET must be used with OPENCL shaders`

Reply:
Ok, I'll add an error.